bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -1607,6 +1612,7 @@ void PreProcessor::handleReqPreProcessingJob(const PreProcessRequestMsgSharedPtr
       preProcessReqMsg->requestLength(),
       preProcessReqMsg->requestBuf(),
       std::string(preProcessReqMsg->requestSignature(), preProcessReqMsg->requestSignatureLength()),
+      preProcessReqMsg->PrimaryblockID(),
```

Comment:
As we pass too many parameters from preProcessReqMsg here, it is worth passing the whole preProcessReqMsg message instead.